src/MediaRatings1L.java

```diff
@@ -0,0 +1,178 @@
+import java.util.LinkedList;
+import java.util.Queue;
+
+/**
+ * MediaRatings represented as a java.util.Map with implementations of primary
+ * methods.
+ *
+ * @convention $this.Pair.media() != null and $this.Pair.rating() >= 0 and
+ *             $this.Pair.rating() <= 5 and MediaRating holds the string and
+ *             integer pair which is held in a queue
+ *
+ * @correspondence this = entries($this.rep)
+ *
+ */
+public class MediaRatings1L extends MediaRatingsSecondary {
+
+    /*
+     * Private Members
+     */
+
+    /**
+     * Elements included in {@code this}.
+     */
+    private final class MediaRating {
+
+        /**
+         * name of media.
+         */
+        private String media;
+
+        /**
+         * value of rating.
+         */
+        private int rating;
+    }
+
+    /**
+     * Representation of this.
+     */
+    private Queue<MediaRating> rep;
+
+    /**
+     * Finds pair with media and moves it to the front of q.
+     *
+     * @param q
+     *            the Queue to be searched
+     * @param media
+     *            the string to be searched for
+     * @updates q
+     * @ensures <pre>
+     * perms(q, #q)  and MediaRating where MediaRating.media = media is in the front
+     * </pre>
+     */
+    private static void moveToFront(Queue<MediaRating> q, String media) {
+        boolean check = false;
+        for (int i = 0; i < q.size(); i++) {
```

Comment:
You might be able to simplify your code here a little. Queues have a rotate method. Usually the way I recommend writing it is like this:
```java
while (!q.front().media.equals(media)) {
    q.rotate(1);
}
```
There's no requires, so you might also need a counter. 
```java
int i = 0;
while (i < q.length() && !q.front().media.equals(media)) {
    q.rotate(1);
    i++;
}
```

Reply:
I don't know if Java queues have a rotate method. Feel free to disregard. I'm leaving the comment anyway just to show another way of writing the code.